doc/development.md

```diff
@@ -89,4 +89,205 @@ merged, we need to test them on all supported distributions and
 architectures.
 
 When you push patches to GitHub, CI will run its tests according to the
-configuration in the `.github/workflows/ci.yml` file.
\ No newline at end of file
+configuration in the `.github/workflows/ci.yml` file.
+
+## Coding guidelines
+
+The coding conventions in Vdsm are generally aligned with the
+[PEP 8](https://peps.python.org/pep-0008/) standard guidelines.
+Code is linted with `pylint` based in the `/.pylintrc` file.
+You can lint the code before submitting a patch by doing:
+
+    make lint
+
+However, some code conventions are not explicitly enforced by the linter due
+to legacy code and inconsistent styles. But new code shall adhere to those
+rules, and old code, should be updated as possible when patched.
+
+In this section, we will cover some of the most important conventions and
+preferences for contributing to Vdsm.
+
+### Imports
+
+First import standard library modules, then 3rd party modules, and finally
+project modules.
+Separate standard library modules and 3rd party modules with a blank line.
+Keep alphabetical order within each block.
+
+Less specific imports:
+
+    from vdsm.foo import baz
+
+Should come before more specific imports like:
+
+    from vdsm.foo.bar import baz
+
+Accepted, but better to avoid:
+- Relative name imports
```

Comment:
I think that absolute imports are more readable, makes easier to locate the imported modules.

At least Python 3 enforces explicit relative imports, which deals with the more problematic case. But still, I don't see a good reason to use relative imports either. The structure of the project is not as complex as to make it really appealing. In any case, in the guidelines they are not forbidden, but I think it is fine to discourage them.

Anyway, if you still think I should remove the relative import from the list to avoid, I'll do it, I don't really have a strong preference.